supply-chain/io/src/lib.rs

```diff
@@ -0,0 +1,104 @@
+#![no_std]
+
+use gstd::{prelude::*, ActorId};
+use primitive_types::U256;
+
+#[derive(Encode, Decode, TypeInfo)]
+pub struct InitSupplyChain {
+    pub producers: BTreeSet<ActorId>,
+    pub distributors: BTreeSet<ActorId>,
+    pub retailers: BTreeSet<ActorId>,
+
+    pub ft_program_id: ActorId,
+    pub nft_program_id: ActorId,
+}
+
+#[derive(Encode, Decode, TypeInfo)]
+pub enum SupplyChainAction {
+    Produce { name: String, notes: String },
+    PutUpForSaleByProducer { item_id: U256, price: u128 },
+    PurchaseByDistributor { item_id: U256, delivery_time: u64 },
+    ShipByProducer(U256),
+    ReceiveByDistributor(U256),
+    ProcessByDistributor(U256),
+    PackageByDistributor(U256),
+    PutUpForSaleByDistributor { item_id: U256, price: u128 },
+    PurchaseByRetailer { item_id: U256, delivery_time: u64 },
+    ShipByDistributor(U256),
+    ReceiveByRetailer(U256),
+    PutUpForSaleByRetailer { item_id: U256, price: u128 },
+    PurchaseByConsumer(U256),
+    GetItemInfo(U256),
```

Comment:
I think it would be better to leave it in both actions and metastate (will add later), so that other smart contracts interacting with supply chain can find out a state of an item.

Reply:
Similarly, other contracts will be able use metastate for this purpose, why not?